iroh-p2p/src/node.rs

```diff
@@ -38,38 +38,22 @@ use crate::keys::{Keychain, Storage};
 use crate::providers::Providers;
 use crate::rpc::{P2p, ProviderRequestKey};
 use crate::swarm::build_swarm;
+use crate::GossipsubEvent;
 use crate::{
     behaviour::{Event, NodeBehaviour},
     rpc::{self, RpcMessage},
     Config,
 };
 
 #[allow(clippy::large_enum_variant)]
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, Serialize, Deserialize)]
```

Comment:
Why does this need to be serializable now? The new rpc message just serializes GossipsubEvent, not this.

Reply:
Correct as well, removed those bits and everything passes tests.